README.md

```diff
@@ -58,6 +58,98 @@ running Go 1.9 or higher.
+
+### The easy way
+
+This repo contains a tool to generate the SDK, which depends on the golang tags from the readme files in the Azure REST API specs repo. The tool assumes you have an [Azure REST API specs](https://github.com/Azure/azure-rest-api-specs) clone. This will generate a single service, including all its API versions.
+
+1. `tools/generator`
+
+1. `go install`
+
+1. Add `GOPATH/bin` to your `PATH`, in case it was not already there.
+
+1. Call the generator tool like this...
+
+``` cmd
+generator –r –v –l=logs/output/folder –version=version path/to/your/swagger/repo/clone/specification/service
+```
+
+ For example
+
+``` cmd
+generator –r –v –l=temp –version=v11.2.0-beta C:/azure-rest-api-specs/specification/network
+```
+
+### Generate the complete SDK
+
+All services, all API versions.
+
+1. Call the generator tool like this...
+
+``` cmd
+generator –r –v –l=logs/output/folder –version=version path/to/your/swagger/repo/clone
+```
+
+ For example
+
+``` cmd
+generator –r –v –l=temp –version=v11.2.0-beta C:/azure-rest-api-specs
+```
+
+### Generate using AutoRest directly
+
+1. Call autorest with the following arguments...
+
+``` cmd
+autorest path/to/readme/file --go --go-sdk-folder='your/gopath/src/github.com/Azure/azure-sdk-for-go' --tag=choose/a/tag/in/the/readme/file --package-version='version' --user-agent='Azure-SDK-For-Go/version services'
+```
+
+For example...
+
+``` cmd
+autorest C:/azure-rest-api-specs/specification/advisor/resource-manager/readme.md --go --go-sdk-folder='C:/goWorkspace/src/github.com/Azure/azure-sdk-for-go' --tag=package-2016-07-preview --package-version='v11.2.0-beta' --user-agent='Azure-SDK-For-Go/v11.2.0-beta services'"
+```
+
+- If you are looking to generate code based on a specific swagger file, you can replace path/to/readme/file with `--input-file=path/to/swagger/file`.
+- If the readme file you want to use as input does not have golang tags yet, you can call autorest like this...
+
+
+``` cmd
+autorest path/to/readme/file --go --license-header=MICROSOFT_APACHE_NO_VERSION --namespace=packageName --output-folder=your/gopath/src/github.com/Azure/azure-sdk-for-go/services/serviceName/mgmt/APIversion/packageName --package-version=version --user-agent='Azure-SDK-For-Go/version services' --clear-output-folder --can-clear-output-folder --tag=choose/a/tag/in/the/readme/file
+```
+
+For example
+
+``` cmd
+autorest https://raw.githubusercontent.com/Azure/azure-rest-api-specs/current/specification/network/resource-manager/readme.md --go --license-header=MICROSOFT_APACHE_NO_VERSION --namespace=network --output-folder=C:/goWorkspace/src/github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network --package-version=v11.2.0-beta --clear-output-folder --can-clear-output-folder --tag=package-2017-09
+```
+
+1. Run `go fmt` on the generated package folder.
+
+1. To make sure the SDK has been generated correctly, also run `golint`, `go build` and `go vet`.
+
+## Include a new package in the SDK
```

Comment:
This is a great section to include, it definitely belongs with the list I typed out above!